src/render-delaying-extensions.js

```diff
@@ -44,9 +44,15 @@ const LOAD_TIMEOUT = 3000;
  */
 export function waitForExtensions(win) {
   const extensions = includedExtensions(win);
-
-  if (extensions.length) {
-    return timer.timeoutPromise(LOAD_TIMEOUT, Promise.all(extensions));
+  const promises = extensions.map((extension) => {
+    return timer.timeoutPromise(
+      LOAD_TIMEOUT,
+      getServicePromise(win, extension),
+      `Render timeout waiting for ${extension} to load.`
+    );
+  });
+  if (promises.length) {
+    return Promise.all(promises);
```

Comment:
Is returning implicit `undefined` right thing here? `undefined` is also not declared in the `@return` jsdoc. Generally, my preference, if `@return` is present, to always terminate with `return` statement, even if it's `return undefined;`. Otherwise it looks a bit like a bug.